custom/enikshay/case_utils.py

```diff
@@ -203,15 +201,24 @@ def get_open_drtb_hiv_case_from_episode(domain, episode_case_id):
 
 def get_open_episode_case_from_person(domain, person_case_id):
     """
-    Gets the first open 'episode' case for the person
+    Gets the first most recently open 'episode' case for the person
 
     Assumes the following case structure:
     Person <--ext-- Occurrence <--ext-- Episode
 
     """
-    return get_open_episode_case_from_occurrence(
-        domain, get_open_occurrence_case_from_person(domain, person_case_id).case_id
-    )
+    episode_cases = get_all_episode_cases_from_person(domain, person_case_id)
+    valid_cases = [case for case in episode_cases if (
+        case.dynamic_case_properties().get('referral_status') not in [
+            'invalid_episode',
+            'duplicate',
+            'invalid_registration'
+        ]
+    )]
+    if not valid_cases:
+        return None
+    else:
+        return sorted(valid_cases, key=(lambda case: case.opened_on))[0]
```

Comment:
i think this will sort in the opposite order. its supposed to be the latest opened case right?